test/test_tags.rb

```diff
@@ -7,21 +7,22 @@ def setup
     FileUtils.mkdir_p("tmp")
   end
 
+  # rubocop:disable Metrics/AbcSize
   def create_post(content, override = {}, converter_class = Jekyll::Converters::Markdown)
     site = fixture_site({ "highlighter" => "rouge" }.merge(override))
 
     site.posts.docs.concat(PostReader.new(site).read_posts("")) if override["read_posts"]
     CollectionReader.new(site).read if override["read_collections"]
     site.read if override["read_all"]
-
     info = { :filters => [Jekyll::Filters], :registers => { :site => site } }
     @converter = site.converters.find { |c| c.instance_of?(converter_class) }
     payload = { "highlighter_prefix" => @converter.highlighter_prefix,
                 "highlighter_suffix" => @converter.highlighter_suffix, }
 
-    @result = Liquid::Template.parse(content).render!(payload, info)
-    @result = @converter.convert(@result)
+    site.send(:set_up_link_registry)
```

Comment:
The fact that we're having to use `send` here is setting off warning bells in my head. Maybe the link registry should be a separate object?